### PR TITLE
handle several stages of filters for licenses

### PR DESCRIPTION
**Pull request by ZouhourArrow**

### CODE REVIEW

File: src/licenses/entities/license/licenseFindResult.ts [hunk 1]
```diff
@@ -164,6 +174,9 @@ export type LicenceFindDataKeywords = {
   [LicenseFindResultFields.COLUMN_RESELLER_NAME]?: DataKeywords;
   [LicenseFindResultFields.COLUMN_RESELLER_REF]?: DataKeywords;
   [LicenseFindResultFields.COLUMN_SEAT]?: DataKeywords;
+  [LicenseFindResultFields.COLUMN_SECURITY]?:
+    | SecurityFindResultDataKeywords
+    | undefined;
```
Comment (sabirab2) [inline src/licenses/entities/license/licenseFindResult.ts:179]: You can just do:  ` [LicenseFindResultFields.COLUMN_SECURITY]?: SecurityFindResultDataKeywords`  the '?' indicates that the parameter is optional (could be undefined)

File: src/licenses/entities/license/licenseFindResult.ts [hunk 2]
```diff
@@ -207,6 +220,9 @@ export type LicenceFindDataSortParameters = {
   [LicenseFindResultFields.COLUMN_RESELLER_NAME]?: SortParameters;
   [LicenseFindResultFields.COLUMN_RESELLER_REF]?: SortParameters;
   [LicenseFindResultFields.COLUMN_SEAT]?: SortParameters;
+  [LicenseFindResultFields.COLUMN_SECURITY]?:
```
Comment (sabirab2) [inline src/licenses/entities/license/licenseFindResult.ts:223]: Same here:  ` [LicenseFindResultFields.COLUMN_SECURITY]?: SecurityFindResultDataSortParameters`

File: src/licenses/entities/license/licenseFindResult.ts [hunk 3]
```diff
@@ -250,6 +266,9 @@ export type LicenceFindDataFiltersParameters = {
   [LicenseFindResultFields.COLUMN_RESELLER_NAME]?: FiltersParameters;
   [LicenseFindResultFields.COLUMN_RESELLER_REF]?: FiltersParameters;
   [LicenseFindResultFields.COLUMN_SEAT]?: FiltersParameters;
+  [LicenseFindResultFields.COLUMN_SECURITY]?:
+    | SecurityFindResultDataFiltersParameters
```
Comment (sabirab2) [inline src/licenses/entities/license/licenseFindResult.ts:270]: Same here:  ` [LicenseFindResultFields.COLUMN_SECURITY]?: SecurityFindResultDataFiltersParameters`

File: src/licenses/entities/license/licenseFindResult.ts [hunk 4]
```diff
@@ -403,6 +423,13 @@ export class LicenseFindResult extends AbstractEntity<LicenseFindResultData> {
     this.#resellerName = data[LicenseFindResultFields.COLUMN_RESELLER_NAME];
     this.#resellerRef = data[LicenseFindResultFields.COLUMN_RESELLER_REF];
     this.#seat = data[LicenseFindResultFields.COLUMN_SEAT];
+    const security: SecurityFindResultData = {
+      [SecurityFindResultFields.COLUMN_ACTIVE_FRAUD_EVENTS]:
+        data?.[LicenseFindResultFields.COLUMN_SECURITY]?.[
+          SecurityFindResultFields.COLUMN_ACTIVE_FRAUD_EVENTS
+        ] ?? null,
```
Comment (sabirab2) [inline src/licenses/entities/license/licenseFindResult.ts:430]:
```suggestion
        ],
```

File: src/licenses/entities/license/securityFindResult.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export type SecurityFindResultDataSortParameters = {
+  [SecurityFindResultFields.COLUMN_ACTIVE_FRAUD_EVENTS]?: SortParameters;
+};
+
+export type SecurityFindResultDataFiltersParameters = {
+  [SecurityFindResultFields.COLUMN_ACTIVE_FRAUD_EVENTS]?: FiltersParameters;
+};
+
+export class SecurityFindResult extends AbstractEntity<SecurityFindResultData> {
+  readonly #activeFraudEvents: number | null | undefined;
```
Comment (sabirab2) [inline src/licenses/entities/license/securityFindResult.ts:29]:
```suggestion
  readonly #activeFraudEvents?: number | null;
```

File: src/licenses/licensesClient.ts [hunk 6]
```diff
@@ -617,4 +577,99 @@ export class LicensesClient extends AbstractRestfulClient {
 
     return await this.put(payload, parameters);
   }
+  private createFilters(
+    parameters: string | BaseParameters<unknown, unknown>,
+    keyParent: string,
```
Comment (sabirab2) [inline src/licenses/licensesClient.ts:582]: Create a new type `KeyParent`:
```ts
type KeyParent = 'license' | 'offer';
```
```suggestion
    keyParent: KeyParent,
```

File: src/licenses/licensesClient.ts [hunk 7]
```diff
@@ -617,4 +577,99 @@ export class LicensesClient extends AbstractRestfulClient {
...
+
+  private createKeywords(
+    parameters: string | LicenseKeywordsParameters,
+    keyParent: string,
+  ): LicenseKeywordsParameters {
+    let appropriateParameters;
+
+    if (typeof parameters === 'object') {
+      appropriateParameters =
+        keyParent === 'license' ? parameters.license : parameters.offer;
```
Comment (sabirab2) [inline src/licenses/licensesClient.ts:632]:
```suggestion
        parameters[keyParent];
```

File: src/licenses/licensesClient.ts [hunk 8]
```diff
@@ -617,4 +577,99 @@ export class LicensesClient extends AbstractRestfulClient {
...
+    } else {
+      appropriateParameters = parameters;
+    }
+
+    return Object.entries(
+      appropriateParameters !== null && appropriateParameters !== void 0
```
Comment (sabirab2) [inline src/licenses/licensesClient.ts:638]:
```suggestion
      appropriateParameters !== null && appropriateParameters !== undefined
```

**Review by sabirab2 — CHANGES_REQUESTED** [review]

**Comment by fdianoux** [comment]
Remove the package-lock.json file please

**Pull request merged**